packages/dds/tree/src/test/simple-tree/unhydratedFlexTreeFromInsertable.spec.ts

```diff
@@ -1012,44 +1010,78 @@ describe("unhydratedFlexTreeFromInsertable", () => {
+				): UnhydratedFlexTreeNode[] => {
+					const s: string = context === "UseGlobalContext" ? "Global" : "Local";
+					log.push(s);
+					return [unhydratedFlexTreeFromInsertable(s, schemaFactory.string)];
+				},
+			);
+			const field = createFieldSchema(FieldKind.Identifier, schemaFactory.string, {
+				defaultProvider: defaultIdentifierProvider,
 			});
 
-			const tree = {};
+			class Schema extends schemaFactory.object("object", {
+				a: field,
+			}) {}
 
-			const actual = unhydratedFlexTreeFromInsertable(tree, schema);
-			const dummy = hydrate(schema, {});
-			const dummyContext = getKernel(dummy).context.flexContext;
-			assert(dummyContext.isHydrated());
-			// Do the default allocation using this context
-			const context: FlexTreeHydratedContextMinimal = {
-				checkout: dummyContext.checkout,
-				nodeKeyManager,
-			};
+			{
+				const node = new Schema({});
+				assert.deepEqual(log, []);
+				const id = node.a;
+				assert.deepEqual(log, ["Global"]);
+				assert.equal(id, "Global");
+				log.length = 0;
+			}
+			{
+				const node = new Schema({});
+				assert.deepEqual(log, []);
+				hydrate(Schema, node);
+				assert.deepEqual(log, ["Local"]);
+				const id = node.a;
+				assert.deepEqual(log, ["Local"]);
+				assert.equal(id, "Local");
+			}
+		});
+
+		// This checks that identifiers are populated lazily from the view ensuring they can be compressed.
+		// This ensure the public facing part of this is working end to end.
+		it("Populates identifier field lazily with the view's compressor when inserting", () => {
+			const schemaFactory = new SchemaFactory("test");
+			class Schema extends schemaFactory.object("object", {
+				a: schemaFactory.identifier,
+			}) {}
 
-			prepareContentForHydration([actual], () => {}, context);
+			// Create the node, with the uninitialized identifier field.
+			// If accessed now, it would allocate a random v4 uuid.
+			const node = new Schema({});
 
-			const expected: MapTree = {
-				type: brand("test.object"),
-				fields: new Map<FieldKey, MapTree[]>([
-					[
-						brand("a"),
-						[
-							{
-								type: brand(stringSchema.identifier),
-								value: nodeKeyManager.getId(0),
-								fields: new Map(),
-							},
-						],
-					],
-				]),
-			};
+			const view = getView(
+				new TreeViewConfiguration({ schema: SchemaFactory.optional(Schema) }),
+				{ idCompressor: createSnapshotCompressor() },
+			);
 
-			assert.deepEqual(deepCopyMapTree(actual), expected);
+			view.initialize(undefined);
+
+			// This should populate the identifier using the view's compressor.
+			view.root = node;
+
+			// If working correctly, the identifier is allocated from the view snapshot compressor and should be deterministic.
+			// If instead allocated early without using the view, it will be a random v4 uuid and fail this check.
+			// Due to some other source of id allocation (like revision tags) there may be a small number of other ids allocated resulting in the last digit not being 1.
```

Comment:
```suggestion
			// Due to some other source of id allocation (like revision tags) there may be a small number of other ids allocated resulting in the last digit not being 0.
```

Reply:
I have added a reference to snapshotSessionId to the comment to help clarify this.